98_exercises.Rmd

```diff
@@ -291,24 +291,87 @@ See the OMA book for examples.
+5. Which and how many assays are contained by each TreeSE?
+6. Store the assays of each TreeSE into a variable named `<TreeSE name>_assays`
+7. Write a csv file for each assays object with the `write.csv` function
+8. Besides experiments, MAEs also include a sampleData and a sampleMap, which
+   are accessible with `colData(mae)` and `sampleMap(mae)`, respectively.
+   Save also each of these two elements into a csv file.
+
+Well done! You just splitted a MAE into its components and stored them as csv files.
+[This script](https://github.com/JuliaTurkuDataScience/MicrobiomeAnalysis.jl/blob/main/src/assets/XO_preprocess.R)
+shows a possible approach. Next, we will try to reconstruct the same MAE
+from the files you created. Make sure you know their names and location!
+
+1. Read in the csv files containing assays with `read.csv` and save each
+   of them into a variable named `<assay name>_assays`
+2. Create one TreeSE from each assays object with the `TreeSummarizedExperiment`
+   function. Note that the function also requires the arguments for rowData
+   and colData, to which you can give the rownames and colnames of the
+   assays object, respectively
+3. Read in the sampleData and the sampleMap and store them into the
+   variables `sample_data` and `sample_map`, respectively
+4. Combine the TreeSEs, the sampleData and the sampleMap into a MAE named `mae`
+   with the `MultiAssayExperiment` function, where the first argument is an
+   `ExperimentList` of your TreeSEs, the second is colData and the third is
+   sampleMap
+5. Make sure that the MAE experiments are identical to the original TreeSEs.
+
+Good job! Now you are aware of how MAEs are built and we can proceed to some
+explorative exercises.
 
 ### Multi-omic data exploration
 
-1. Load experimental dataset from microbiomeDataSets (e.g., HintikkaXOData).
-2. Analyze correlations between experiments. (Taxa vs lipids, Taxa vs biomarkers, Lipids vs biomarkers)
-3. Agglomerate taxa data.
-4. Apply CLR to taxa data, apply log10 to lipids and biomarkers.
-5. Perform cross-correlation analyses and visualize results with heatmaps. (Use Spearman coefficients)
-6. Is there significant correlations? Interpret your results.
-
-Useful functions: pheatmap, ComplexHeatMap::Heatmap, ggplot, transformCounts, testExperimentCrossAssociation
+Here we learn some explorative methods and apply to the HintikkaXOData. If you
+solved the previous exercise, you can use the MAE you created and skip the first
+step.
+
+1. Store the HintikkaXOData MAE into a variable named `mae`
+2. Which experiments make up the MAE? How many samples and features are contained
+   by each experiment? You can check that with `dim`, `nrow` and `ncol`
+3. Agglomerate the microbiota experiment by Genus and store it into a new TreeSE
+   named `microbiota_genus`. How many features remain after agglomerating?
+4. Add `microbiota_genus` as a new experiment to `mae`. Note that new experiments
```

Comment:
These are more advanced, I would split this into its own exercise.

Reply:
In fact this is not good example because we are suggesting to store different levels of taxonomic tables into altExp, see Table 3.1 https://microbiome.github.io/OMA/containers.html#data-science-framework

Reply:
You could ask to add the genus level table into a new altExp within the microbiome experiment in mae..?

Reply:
For adding new MAE components, how about just construcing one MAE with two components first (microbiota + metabolites?) and then adding third in a separate step?